src/selectors/getSortedRank.ts

```diff
@@ -7,7 +8,7 @@ import getSortPreference from './getSortPreference'
 import thoughtToPath from './thoughtToPath'
 
 /** Calculates the rank for a given index in a sorted array of thoughts. */
-const calculateRank = (thoughts: { rank: number }[], index: number): number => {
+const calculateRank = (thoughts: Thought[], index: number, id?: string): number => {
```

Comment:
I don't love the new API of this function. That third parameter is very awkward. I think this needs some work, or needs to be addressed higher up.